src/engine/CodeValidator.v3

```diff
@@ -1266,6 +1351,15 @@ class CodeValidator(extensions: Extension.set, limits: Limits, module: Module, e
 			if (!ValueTypes.isAssignable(ct[j], p[j])) err_atpc().TypeMismatchInCatch(i, p[i], ct[i]);
 		}
 	}
+	def checkContHandle(i: int, tag: TagDecl, target: ControlEntry) {
+		var p = labelArgs(target);
+		if (p == null) return;
+		var ct = Arrays.append(ValueType.Ref(true, HeapType.ANY), tag.fields);
```

Comment:
Is `HeapType.ANY` right here? I think this may be a `ref cont`?

Reply:
Sorry I forgot to ask about this. Currently it seems that (I'm not sure if this is really the case, but this behavior happened in the test cases I'm using) when reading a `(ref $cont)` as the result of a block, the parser interprets it as a `Ref(..., HeapType.ANY)`. I'm guessing this might be because the bytecode format of `ref` is a generic index into `heaptypes`. This results in the top value in `labelArgs(block)` is a `ref any`. I used `Ref(true, HeapType.ANY)` as a placeholder.

Reply:
Followup: this seems to be caused by `readHeapType` returning `ANY` on reading `ref cont`. I must've missed this in my previous PR. Will fix this.